src/NuGet.Core/NuGet.Packaging/Signing/Authoring/Signer.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Remove or add signature package metadata.
+    /// </summary>
+    public sealed class Signer
+    {
+        private readonly ISignedPackage _package;
+        private readonly SigningSpecificationsV1 _specifications = SigningSpecifications.V1;
```

Comment:
wouldn't it be better if this variable is of type `SigningSpecifications` ? Therefore if the version changes only the value will change but not the type

Reply:
It uses things that are v1 specific currently. I tried to avoid making a pattern that is too general for now since no matter what it will end up looking exactly like v1 since there is no v2.